#167749950 User email verification

ibraheem-adeyemo opened the pull request:
#### What does this PR do?
- It verifies users email

#### Description of task to be completed
- Have verifyEmail function that sends mail to a user
- Have api/v1/verifyEmail/:token route that user reach when clicking the link 
- It updates isVerified column to true

#### How should this be manually tested?
After cloning the repo cd into it and run npm install and npm test
- using the postman test 
- make signup request with the firstname, lastname, email, password.
- check the mail and click the link.

Key: content-type value: application/json

Key: Authorization value: JWT TOKEN

#### What are the relevant pivotal tracker stories
[#167749950](https://www.pivotaltracker.com/story/show/167749950)

Naimaoye reviewed (commented):
You did a good job on this. I dont think I have any request for now. Well done!

hustlaviola commented on .travis.yml:
```diff
@@ -9,6 +9,11 @@ before_script:
 node_js:
 - 'stable'
 
+env:
```
You can remove the environment variables. It has been added on Travis

ibraheem-adeyemo replied:
Implemented

hustlaviola commented on src/middlewares/userValidations.js:
```diff
@@ -16,7 +16,7 @@ const validateUser = path => (req, res, next) => {
     const schema = _.get(Schemas, path);
     const response = Joi.validate(user, schema, { abortEarly: false });
-    const errors = Helper.buildErrorResponse(response);
+    const errors = Helper.handleValidation(response);
```
You should revert the changes in this file

ibraheem-adeyemo replied:
Done!

hustlaviola commented on src/middlewares/userValidations.js:
```diff
@@ -31,8 +31,7 @@ const validateUser = path => (req, res, next) => {
 const emailExists = (req, res, next) => {
-  let { email } = req.body;
-  email = email.trim().toLowerCase();
+  const { email } = req.body;
```
`email` has been trimmed and changed to lowercase.

ibraheem-adeyemo replied:
Changes implemented

hustlaviola commented on src/routes/index.js:
```diff
@@ -0,0 +1,11 @@
+import express from 'express';
```
This file has been deleted already. You should remove it

ibraheem-adeyemo replied:
All right 

Removed

hustlaviola commented on src/utils/Helper.js:
```diff
@@ -21,14 +54,14 @@ export default class Helper {
    * @memberof Helper
    */
-  static buildErrorResponse(data) {
+  static handleValidation(data) {
```
This method's name does not need to change. Please revert it.

ibraheem-adeyemo replied:
Changes implemented

hustlaviola reviewed (changes requested):
Awesome @Ibraheem-adeyemo. However, some changes were not necessary. It probably occurred when you were rebasing, maybe you accepted `incoming change` where you should have accepted the `current change`. Please check the inline comments to find the errored areas. You also need to write some test cases for this endpoint and document with Swagger. Good job

ibraheem-adeyemo committed:
feat(verify email):verify user email

- Add isVerify column to the model
- Add nodemailer configuration
- Add nodemailer configuration
- Add verify route to complete thet verification

[Finishes #16774995]

hustlaviola commented on src/middlewares/userValidations.js:
```diff
@@ -31,8 +31,7 @@ const validateUser = path => (req, res, next) => {
 const emailExists = (req, res, next) => {
-  let { email } = req.body;
-  email = email.trim().toLowerCase();
+  const { email } = req.body;
```
This need to be trimmed and converted to lower case.
```suggestion
  let { email } = req.body;
  email = email.trim().toLowerCase();
```

hustlaviola commented on .eslintrc.json:
```diff
@@ -41,5 +41,4 @@
     }
-}
-
+ }
```
You can add a newline here

ldonjibson reviewed (approved):
Good @Ibraheem-adeyemo

hustlaviola commented on src/database/models/user.js:
```diff
@@ -1,22 +1,11 @@
-    },
+    firstName: { type: DataTypes.STRING, allowNull: false },
+    lastName: { type: DataTypes.STRING, allowNull: false },
+    email: { type: DataTypes.STRING, allowNull: false },
```
```suggestion
    email: { type: DataTypes.STRING, allowNull: false, unique: true },
```

hustlaviola commented on src/database/migrations/20190901204457-create-user.js:
```diff
@@ -0,0 +1,40 @@
+    },
+    email: {
+      type: Sequelize.STRING,
+      allowNull: false
```
```suggestion
      allowNull: false,
      unique: true
```

daviddamilola reviewed (approved):
LGTM
